Compilation fails with updated `maven-compiler-plugin`

### Context

As part of the `smallrye-parent` pull request CI, we build and test `smallrye-mutiny`.

### Description

When upgrading `maven-compiler-plugin` to `3.12.0`, the Mutiny build fails:

```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for SmallRye Mutiny - Parent 999-SNAPSHOT:
[INFO] 
[INFO] SmallRye Mutiny - Parent ........................... SUCCESS [  2.666 s]
[INFO] SmallRye Mutiny - Testing utilities ................ SUCCESS [  3.772 s]
[INFO] SmallRye Mutiny - Core library ..................... SUCCESS [08:32 min]
[INFO] SmallRye Mutiny - Tests against the Reactive Streams TCK SUCCESS [03:40 min]
[INFO] SmallRye Mutiny - Integration with RxJava 3 ........ SUCCESS [  6.920 s]
[INFO] SmallRye Mutiny - Integration with Reactor ......... SUCCESS [ 12.057 s]
[INFO] SmallRye Mutiny - Integration with SmallRye Context Propagation SUCCESS [  2.492 s]
[INFO] SmallRye Mutiny - MicroProfile Reactive Streams Operators implementation SUCCESS [01:24 min]
[INFO] SmallRye Mutiny - MicroProfile Reactive Streams Operators 3.x implementation SUCCESS [01:22 min]
[INFO] SmallRye Mutiny - Kotlin Extensions ................ FAILURE [  0.442 s]
[INFO] SmallRye Mutiny - Documentation .................... SKIPPED
[INFO] SmallRye Mutiny - BOM .............................. SKIPPED
[INFO] SmallRye Mutiny - Math operators ................... SKIPPED
[INFO] SmallRye Mutiny - Workshop examples ................ SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  15:28 min
[INFO] Finished at: 2023-12-19T05:52:08Z
[INFO] ------------------------------------------------------------------------
Error:  Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.12.0:compile (default-compile) on project mutiny-kotlin: Fatal error compiling: basedir /home/runner/work/smallrye-parent/smallrye-parent/smallrye-mutiny/kotlin/target/generated-sources/annotations does not exist -> [Help 1]
Error:  
Error:  To see the full stack trace of the errors, re-run Maven with the -e switch.
Error:  Re-run Maven using the -X switch to enable full debug logging.
Error:  
Error:  For more information about the errors and possible solutions, please read the following articles:
Error:  [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
Error:  
Error:  After correcting the problems, you can resume the build with the command
Error:    mvn <args> -rf :mutiny-kotlin
Error: Process completed with exit code 1.
```

### Additional details

This is the link to the build failure in CI: https://github.com/smallrye/smallrye-parent/actions/runs/7257824786/job/19772186757?pr=461

See also smallrye/smallrye-parent#461.

--- a/kotlin/pom.xml
+++ b/kotlin/pom.xml
@@ -72,6 +72,14 @@
         <testSourceDirectory>${project.basedir}/src/test/kotlin</testSourceDirectory>
         <plugins>
             <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <skip>true</skip>
+                    <skipMain>true</skipMain>
+                </configuration>
+            </plugin>
+            <plugin>
                 <groupId>org.jetbrains.kotlin</groupId>
                 <artifactId>kotlin-maven-plugin</artifactId>
                 <version>${kotlin.version}</version>
@@ -147,5 +155,5 @@
             </build>
         </profile>
     </profiles>
-    
+
 </project>

